front_end/panels/react_devtools/ReactDevToolsView.ts

```diff
@@ -94,38 +91,79 @@ export class ReactDevToolsViewImpl extends UI.View.SimpleView {
       send: (event, payload): void => this.sendMessage(event, payload),
     };
 
-    // To use the custom Wall we've created, we need to also create our own "Bridge" and "Store" objects.
-    this.bridge = ReactDevTools.createBridge(this.wall);
-    this.store = ReactDevTools.createStore(this.bridge);
-
     // Notify backend if Chrome DevTools was closed, marking frontend as disconnected
-    window.addEventListener('beforeunload', () => this.bridge.shutdown());
+    window.addEventListener('beforeunload', () => {
+      console.debug('[ReactDevToolsView] beforeunload');
+      this.bridge?.shutdown()
+    });
 
     SDK.TargetManager.TargetManager.instance().addModelListener(
       ReactDevToolsModel,
-      ReactDevToolsModelEvents.MessageReceived,
-      this.onMessage,
+      ReactDevToolsModelEvents.Initialized,
+      this.onInitialized,
       this,
     );
     SDK.TargetManager.TargetManager.instance().addModelListener(
       ReactDevToolsModel,
-      ReactDevToolsModelEvents.Initialized,
-      this.initialize,
+      ReactDevToolsModelEvents.Destroyed,
+      this.onDestroyed,
       this,
     );
     SDK.TargetManager.TargetManager.instance().addModelListener(
-      SDK.RuntimeModel.RuntimeModel,
-      SDK.RuntimeModel.Events.ExecutionContextDestroyed,
-      this.onExecutionContextDestroyed,
+      ReactDevToolsModel,
+      ReactDevToolsModelEvents.MessageReceived,
+      this.onMessage,
       this,
     );
+
+    this.renderLoader();
+  }
+
+  private onInitialized({data: executionContext}: ReactDevToolsInitializedEvent): void {
+    console.debug('[ReactDevToolsView] onInitialized');
+
+    this.clearLoader();
+
+    this.executionContext = executionContext;
+    this.bridge = ReactDevTools.createBridge(this.wall);
+    this.store = ReactDevTools.createStore(this.bridge);
+
     SDK.TargetManager.TargetManager.instance().addModelListener(
-      SDK.RuntimeModel.RuntimeModel,
-      SDK.RuntimeModel.Events.ExecutionContextCreated,
-      this.onExecutionContextCreated,
+      ReactDevToolsModel,
+      ReactDevToolsModelEvents.MessageReceived,
+      this.onMessage,
       this,
     );
 
+    const usingDarkTheme = window.matchMedia('(prefers-color-scheme: dark)').matches;
+    ReactDevTools.initialize(this.contentElement, {
+      bridge: this.bridge,
+      store: this.store,
+      theme: usingDarkTheme ? 'dark' : 'light',
+      canViewElementSourceFunction: () => true,
+      viewElementSourceFunction,
+    });
+  }
+
+  private onDestroyed(): void {
+    console.debug('[ReactDevToolsView] onDestroyed');
+
+    // Unmount React DevTools view
+    this.contentElement.removeChildren();
+
+    SDK.TargetManager.TargetManager.instance().removeModelListener(
+      ReactDevToolsModel,
+      ReactDevToolsModelEvents.MessageReceived,
+      this.onMessage,
+      this,
+    );
```

Comment:
This should be after `bridge.shutdown()` call, so Frontend can notify Backend that bridge is destroyed

Reply:
Oh, but there is no one to notify, because context is already destroyed, can you add a comment explaining this?

Reply:
I don't think we need to add a comment. Just in the previous line we clear the listeners, so there's no point to keeping this since
```
    for (const listener of this.listeners) {
     listener(message);
    }	 
```
wil just do nothing

Reply:
Can you just subscribe once in `constructor`? You won't need to unsubscribe / re-subscribe again in `Initialized` / `Destroyed` events

Reply:
yes. you are right! done.